velox/experimental/cudf/exec/ToCudf.cpp

```diff
@@ -93,30 +103,20 @@ bool CompileState::compile(bool force_replace) {
     return driverFactory_.consumerNode;
   };
 
-  auto isTableScanSupported = [getPlanNode](const exec::Operator* op) {
-    if (!isAnyOf<exec::TableScan>(op)) {
-      return false;
-    }
-    auto tableScanNode = std::dynamic_pointer_cast<const core::TableScanNode>(
-        getPlanNode(op->planNodeId()));
-    VELOX_CHECK(tableScanNode != nullptr);
-    auto const& connector = velox::connector::getConnector(
-        tableScanNode->tableHandle()->connectorId());
-    auto cudfHiveConnector = std::dynamic_pointer_cast<
-        facebook::velox::cudf_velox::connector::hive::CudfHiveConnector>(
-        connector);
-    if (!cudfHiveConnector) {
-      return false;
-    }
-    // TODO (dm): we need to ask CudfHiveConnector whether this table handle is
-    // supported by it. It may choose to produce a HiveDatasource.
-    return true;
-  };
+  const bool isParquetConnectorRegistered =
+      facebook::velox::connector::getAllConnectors().count("test-parquet") > 0;
+  auto isTableScanSupported =
+      [isParquetConnectorRegistered](const exec::Operator* op) {
+        return isAnyOf<exec::TableScan>(op) && isParquetConnectorRegistered &&
+            cudfTableScanEnabled();
+      };
 
-  auto isFilterProjectSupported = [](const exec::Operator* op) {
+  auto isFilterProjectSupported = [getPlanNode](const exec::Operator* op) {
     if (auto filterProjectOp = dynamic_cast<const exec::FilterProject*>(op)) {
-      auto info = filterProjectOp->exprsAndProjection();
-      return ExpressionEvaluator::canBeEvaluated(info.exprs->exprs());
+      auto projectPlanNode = std::dynamic_pointer_cast<const core::ProjectNode>(
+          getPlanNode(filterProjectOp->planNodeId()));
+      const auto exprs = exprsAndProjection(*filterProjectOp, projectPlanNode);
+      return ExpressionEvaluator::canBeEvaluated(exprs);
```

Comment:
This can be simplified so that the exprsAndProjection function above is not needed
```suggestion
      auto const& projectPlanNode = std::dynamic_pointer_cast<const core::ProjectNode>(
          getPlanNode(filterProjectOp->planNodeId()));
      auto const& filterPlanNode = filterProjectOp->filterNode();
   
      return ExpressionEvaluator::canBeEvaluated(projectPlanNode->projections()) &&
                  ExpressionEvaluator::canBeEvaluated(filterPlanNode->filter());
```